iterator.go

```diff
@@ -95,16 +95,20 @@ func (item *Item) Version() uint64 {
 // If you need to use a value outside a transaction, please use Item.ValueCopy
 // instead, or copy it yourself. Value might change once discard or commit is called.
 // Use ValueCopy if you want to do a Set after Get.
-func (item *Item) Value() ([]byte, error) {
+func (item *Item) Value(fn func(val []byte)) error {
```

Comment:
Hey, sorry for the unsolicited feedback. Not sure if you're mimicking another reference API, but in this case I wonder if this function signature would be more convenient:

```diff
- func (item *Item) Value(fn func(val []byte)) error {
+ func (item *Item) Value(fn func(val []byte) error) error {
```

Where the returned error would be forwarded into Value's error. This way it could be used as:

```go
	return item.Value(func(val []byte) error {
		return NewDecoder(bytes.NewReader(val)).Decode(into)
	})
```

Instead of

```go
    var err error
	err = item.Value(func(val []byte) {
		if decodeErr := NewDecoder(bytes.NewReader(val)).Decode(into); decodeErr != nil {
			err = decodeErr
		}
	})
    return err
```

Feel free to disregard this feedback if it's not relevant. :)